benchmarks/dynamo/ci_expected_accuracy/update_expected.py

```diff
@@ -0,0 +1,140 @@
+"""
+Update commited CSV files used as reference points by dynamo/inductor CI.
+
+Currently only cares about graph breaks, so only saves those columns.
+
+Hardcodes a list of job names and artifacts per job, but builds the lookup
+by querying github sha and finding associated github actions workflow ID and CI jobs,
+downloading artifact zips, extracting CSVs and filtering them.
+
+Usage:
+
+1) get a rocksdp API key and save it in ~/rocks_api_key
+2) python benchmarks/dynamo/ci_expected_accuracy.py <sha of pytorch commit that has completed inductor benchmark jobs>
+
+Known limitations:
+- doesn't handle 'retry' jobs in CI, if the same hash has more than one set of artifacts, gets the first one
+"""
+
+import argparse
+import os
+import urllib
+from io import BytesIO
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import pandas as pd
+import requests
+
+
+ARTIFACTS_QUERY_URL = "https://api.usw2a1.rockset.com/v1/public/shared_lambdas/d7052c1e-b96c-44f3-b6cf-bc8fd5c4b7e4"
+
+
+def query_job_sha(repo, sha):
+
+    params = {
+        "parameters": [
+            {
+                "name": "sha",
+                "type": "string",
+                "value": sha,
+            },
+            {"name": "repo", "type": "string", "value": repo},
+        ]
+    }
+
+    r = requests.post(url=ARTIFACTS_QUERY_URL, json=params)
+    data = r.json()
+    return data["results"]
+
+
+def parse_job_name(job_str):
+    return (part.strip() for part in job_str.split("/"))
+
+
+def parse_test_str(test_str):
+    return (part.strip() for part in test_str[6:].strip(")").split(","))
+
+
+S3_BASE_URL = "https://gha-artifacts.s3.amazonaws.com"
+
+
+def get_artifacts_urls(results, suites):
+    urls = {}
+    for r in results:
+        if "inductor" == r["workflowName"] and "test" in r["jobName"]:
+            config_str, test_str = parse_job_name(r["jobName"])
+            suite, shard_id, num_shards, machine = parse_test_str(test_str)
+            workflowId = r["workflowId"]
+            id = r["id"]
+            runattempt = 1  # ? guessing here
```

Comment:
You should be able to get the runattempt from rockset as well (IIRC, it's another column on the jobs/workflows table)

Not all jobs even have an attempt #1. For example, the first run attempt on PRs submitted by external branches will be `runattempt=2`